developer_guides/firmware/llext_modules.rst

```diff
@@ -0,0 +1,78 @@
+************************
+
+SOF has multiple ways to implement loadable modules. LLEXT is one of them.
+With it modules are built as shared or relocatable ELF objects with an addition
+of a cryptographic signature and a manifest. When loaded and instantiated,
+Zephyr LLEXT functionality is used to dynamically resolve module internal as
+well as SOF external code and data references. In the future support for
+inter-module linking will be added.
+
+Accessing the base firmware from LLEXT modules
+**********************************************
+
+LLEXT modules can access all code and data from the base firmware exported,
+using the ``EXPORT_SYMBOL()`` macro. Therefore writing LLEXT modules isn't very
+different from built-in ones.
+
+Implementing LLEXT modules
+**************************
+
+At the moment only modules, implementing the Module Adapter API
+:ref:`apps-comp-world` are supported.
+
+As explained above, LLEXT modules in general look very similar to native SOF
+code, with the only restriction of having no access to not-exported symbols. If
+any such symbols are needed to LLEXT modules, additionally exporting them can be
+considered.
+
+LLEXT modules should also contain a ``.buildinfo`` section, containing a
+``struct sof_module_api_build_info`` object and a ``.module`` section,
+containing a ``struct sof_man_module_manifest`` object. The latter should also
+contain a pointer to a module entry point function, returning a pointer to the
+module's ``struct module_interface`` instance. See
+src/samples/audio/smart_amp_test_ipc4.c for an example.
+
+A TOML configuration file is needed for building of LLEXT modules too. It is
+generated by the C preprocessor at build time from the same components, as would
+be used for a monolithic build. A small additional file to bring components
+together is needed too, similar to
+src/samples/audio/smart_amp_test_llext/llext.toml.h.
+
+Finally an additional CMakeLists.txt is needed similar to
+src/samples/audio/smart_amp_test_llext/CMakeLists.txt. Currently it contains a
+lot of open-coded instructions, later it will be ported to use Zephyr's
+``add_llext_target()`` function.
+
+With that in place, it is also possible to switch between monolithic and modular
+builds by specifying the module as "tristate" in its Kconfig and selecting "m"
+for modular builds, however, for modules, exclusively built as LLEXT objects,
+this might not be necessary. Thus it is possible to implement third party Module
+Adapter drivers, that would be built exclusively as loadable modules.
+
+Note, that a single LLEXT object file can also contain multiple Module Adapter
+modules. In that case an array of ``struct sof_module_api_build_info`` objects
+is needed and the TOML file should contain those multiple module entries too.
+
+Installation
+************
+
+As described in :ref:`intel_debug_introduction` the |SOF| Linux kernel driver
+loads SOF modules, using their UUIDs. For SOF internal modules the process
```

Comment:
What does "using their UUIDs" mean? How does the kernel know which modules to load and which not?

Reply:
@marc-hb Topology specifies UUIDs, the kernel driver also parses the base firmware manifest to see which UUIDs are built-in. For any, that are specified in topology but aren't available in the base firmware, an attempt will be made to load them dynamically. But this is generic SOF architecture information, not LLEXT specific

Reply:
Thanks, I think I found it at `_apps-component-overview`, please add a link to that.

Earlier you mentioned that a single LLEXT file can have multiple modules, in that case which UUID is used for its filename?

Reply:
> Earlier you mentioned that a single LLEXT file can have multiple modules, in that case which UUID is used for its filename?

@marc-hb none. Filenames are created from a common name: "eq_iir," "mixin_mixout," etc. UUIDs are soft links to those files

Reply:
Did you mean "all of them"? As in:
```
  UUID1.bin -> module1and2combined.llext
  UUID2.bin -> module1and2combined.llext
```